pkgs/development/libraries/glibc/locales.nix

```diff
@@ -25,13 +25,18 @@ callPackage ./common.nix { inherit stdenv; } {
   # $TMPDIR as a prefix, meaning that the locale-archive is placed in
   # $TMPDIR/nix/store/...-glibc-.../lib/locale/locale-archive.
   buildPhase =
+    let endianFlag =
+      if stdenv.hostPlatform.isLittleEndian
+      then "--little-endian"
+      else "--big-endian";
+    in
     ''
       mkdir -p $TMPDIR/"${buildPackages.glibc.out}/lib/locale"
 
       echo 'C.UTF-8/UTF-8 \' >> ../glibc-2*/localedata/SUPPORTED
 
       # Hack to allow building of the locales (needed since glibc-2.12)
-      sed -i -e 's,^$(rtld-prefix) $(common-objpfx)locale/localedef,localedef --prefix='$TMPDIR',' ../glibc-2*/localedata/Makefile
+      sed -i -e 's,^$(rtld-prefix) $(common-objpfx)locale/localedef,localedef --prefix='$TMPDIR' ${endianFlag},' ../glibc-2*/localedata/Makefile
```

Comment:
Hmm, it doesn't feel right to add this too to this line. Could it maybe be separated out to prevent putting all kind of logic into this line?